Curric/add sidebar filters

lioneljingles opened the pull request:
## Description
- Added sidebar to curriculum visualiser
- Added radio buttons for threads
- Added radio buttons for year groups
- Updated filtering logic

## How to test

1. Go to https://deploy-preview-1899--oak-web-application.netlify.thenational.academy/teachers/curriculum/geography-primary/units
2. Scroll down to `Units` tab
3. For threads, you should see a title, a selected default radio button "None highlighted" and radio buttons for threads
4. For year groups, you should see a title, a selected default radio button "All" and radio buttons for years
5. Click on a thread
6. You should see the radio button container highlighted
7. You should see the units with the selected thread highlighted
8. Click on a year
9. You should only that year on the unit list on the right

NOTE: full mobile support is not yet in place

## Screenshots

How it should now look:
{screenshots}
<img width="1176" alt="Screenshot 2023-09-14 at 10 37 41" src="https://github.com/oaknational/Oak-Web-Application/assets/421186/611ecbb3-1817-486a-90d0-ee2b34ba387e">
<img width="1176" alt="Screenshot 2023-09-14 at 10 37 51" src="https://github.com/oaknational/Oak-Web-Application/assets/421186/2a09bc87-6501-4cab-bad1-e5864bd082a5">
<img width="1176" alt="Screenshot 2023-09-14 at 10 38 07" src="https://github.com/oaknational/Oak-Web-Application/assets/421186/062ef940-b8da-48a3-8fbb-2cb8a1b920d0">

## Checklist

- [x] Added or updated tests where appropriate
- [x] Manually tested across browsers / devices
- [x] Considered impact on accessibility
- [x] Design sign-off
- [x] Approved by product owner

rightsaidjames commented:
Quite a few discrepancies:
* The Figma designs show a `Clear X` button instead of a 'None highlighted' option, that is only shown when a thread has been highlighted.
* Additionally, the 'None highlighted' option is not in vertical alignment with the other radio buttons.
* The colour of selected radio buttons should be black, not blue, to match the colour of the highlighted tiles.
* In secondary phase curricula, the alphabetical sorting of the year group filter options places years 10 and 11 before years 7-9.
* The text `X units highlighted` is not displayed on the thread filter option label when said thread has been chosen from the sidebar.
* According to the AC, users should be able to click on a selected thread filter option to unselect that thread, as if they had clicked the Clear button. I think this would be okay to leave for post-MVP though, as it's a fairly minor detail.
* Apologies for not documenting this on the ticket, but me and @HelenHarlow [agreed that](https://www.figma.com/file/3cz9fb0ObFDLv1c212OtJv?node-id=811:355599&mode=dev#529321280) visually-hidden prefix text of `Highlighted: ` should be added to cards that are currently highlighted (e.g. `Highlighted: 6 A Monster Within`). This allows screen reader users to benefit from the thread highlighting functionality.
* This wasn't introduced within this ticket, but I've spotted that the year headings have a H4 tag but hierarchically these should be H2s.

Apologies for the laundry list of issues! 

![Screenshot 2023-09-13 at 11 31 22](https://github.com/oaknational/Oak-Web-Application/assets/3018395/1e852be2-6e2a-4997-adc4-5a2c335aff4d)
<img width="842" alt="Screenshot 2023-09-13 at 11 49 53" src="https://github.com/oaknational/Oak-Web-Application/assets/3018395/a1b246a3-7187-4139-91e4-f7774e5e54e7">

rightsaidjames commented on src/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab.tsx:
```diff
@@ -449,36 +468,36 @@ const UnitsTab: FC<UnitsTabProps> = ({ data }) => {
                               {index + 1}
                             </OutlineHeading>
-                            <Heading tag={"h3"} $font={"heading-7"}>
+                            <Box $font={"heading-7"}>
```
The unit title itself should still be a H3 heading, I think.

rightsaidjames commented on src/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab.tsx:
```diff
@@ -449,36 +468,36 @@ const UnitsTab: FC<UnitsTabProps> = ({ data }) => {
-                            <Heading tag={"h3"} $font={"heading-7"}>
+                            <Box $font={"heading-7"}>
+                              {isHighlighted && (
+                                <VisuallyHidden>Highlighted:</VisuallyHidden>
```
It's hard to tell here, but I think we're missing a space between `Highlighted:` and the unit title.

lioneljingles commented:
@rightsaidjames addressed all the issues in your comment except this one (I put a message in the Notion ticket asking for clarification):
> The colour of selected radio buttons should be black, not blue, to match the colour of the highlighted tiles.

Please have a look at the latest build!

rightsaidjames commented:
Outstanding issues:
* As noted above, the unit title should still be a H3 heading.
* As noted above, we need a space after the `Highlighted:` in the visually hidden text element.
* On the lower end of the tablet layout, the unit cards are in two cramped columns that don't make full use of the horizontal space available. As a result this leads to longer unit titles being cut off.
* The `X units highlighted` feature doesn't work properly - it always says `0 units highlighted`
* I would also argue that the filter options in the current mobile layout are too close to the left edge of the screen.

<img width="814" alt="Screenshot 2023-09-14 at 10 02 04" src="https://github.com/oaknational/Oak-Web-Application/assets/3018395/eab0629b-ca5b-4c0e-bd2b-946dce4a599b">
<img width="1679" alt="Screenshot 2023-09-14 at 10 02 13" src="https://github.com/oaknational/Oak-Web-Application/assets/3018395/a7c145a7-a269-4b24-8a7f-e28e4b0c86ce">
<img width="499" alt="Screenshot 2023-09-14 at 10 14 31" src="https://github.com/oaknational/Oak-Web-Application/assets/3018395/b7d94264-87d5-4926-89cf-7dffaee17864">

lioneljingles commented:
@rightsaidjames latest deploy should address outstanding issues (except blue radio button)

rightsaidjames commented on src/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab.tsx:
```diff
@@ -214,129 +290,245 @@ const UnitsTab: FC<UnitsTabProps> = ({ data }) => {
+                            <BrushBorders
+                              color={isHighlighted ? "black" : "white"}
+                            />
+                            <OutlineHeading tag={"div"} $fontSize={24} $mb={12}>
```
This looks like it's a pre-existing issue, but from what I can see in Figma these unit numbers should be 600 font weight instead of 300 (`Heading/Regular/Heading 5`).

rightsaidjames commented:
All previous comments (other than blue/black issue) are resolved or designated as post-MVP - looks much better!

As well as the above comment on the styling of the unit numbers, I've also found a functional bug:

1. Highlight a thread
2. Select a year group (or switch from a specific year group to `All`

**Expected:** The `X units highlighted` count updates based on the currently selected year group (or all year groups)
**Actual:**  The `X units highlighted` count does not update.

<img width="1677" alt="Screenshot 2023-09-14 at 12 47 57" src="https://github.com/oaknational/Oak-Web-Application/assets/3018395/139cb590-a3dc-4f1d-a116-21bf30485340">
<img width="1676" alt="Screenshot 2023-09-14 at 12 48 46" src="https://github.com/oaknational/Oak-Web-Application/assets/3018395/88cbabda-ec94-42b0-bfc8-ab0dc9e89663">

k-huggs commented on src/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab.tsx:
```diff
@@ -179,6 +224,41 @@ const UnitsTab: FC<UnitsTabProps> = ({ data }) => {
+    let count = 0;
+    Object.keys(yearData).forEach((year) => {
+      const data = yearData[year];
+      if (data && (!selectedYear || selectedYear == year)) {
```
=== strict equality needed here

lioneljingles replied:
Done

k-huggs reviewed (approved):
LGTM, one small comment

rightsaidjames commented:
Everything looks good now - matches design (other than mobile optimisation stuff) and functionally everything works as it should. LGTM, assuming @HelenHarlow is happy!

HelenHarlow reviewed (approved):
All looks great from a UI perspective :)
